### DISCUSSION TITLE
Pass product metadata and attributes (from checkout) to orderline

### DISCUSSION THREAD

**Discussion by jha-r** [saleor/saleor#16573] [General] [answered]
The use case I'm working on requires to pass product metadata and attributes to Orderline.  I'm using 3.19 and trying to make changes to the default storefront code.

I can see many fields are passed to orderline e.g. productName, productSku etc; I'm not sure how to add metadata and attributes.

Can you please help what mutation / code needs to be changed in storefront?

Experts any help would be highly appreciated

@patrys @sixthgear @borekb @timuric

**Comment by korycins** [comment, marked as answer]
You can [create an app](https://docs.saleor.io/developer/extending/apps/overview) that would be subscribed to order-created event, then the app can make a request to [update OrderLine's metadata.](https://docs.saleor.io/api-reference/miscellaneous/mutations/update-metadata).
In the case of product attributes, there is no way to assign them directly to the order-line.  You can probably use orderline's metadata to denormalize the attribute details.

**Reply by jha-r** [reply]
Thanks for the quick response.

**Comment by timuric** [comment]
Please take a look at the product personalization guide https://docs.saleor.io/developer/checkout/cookbook#product-personalization

**Reply by jha-r** [reply]
Thanks for the quick response.
